lint/lenNil_checker.go

```diff
@@ -0,0 +1,71 @@
+package lint
+
+//! Detects redundant x!=nil before len(x).
+//
+// @Before:
+// if myMap != nil && len(myMap) == smth {
+//
+// @After:
+// if len(myMap) == smth {
+
+import (
+	"go/ast"
+	"go/token"
+	"go/types"
+)
+
+func init() {
+	addChecker(&lenNilChecker{})
```

Comment:
Should be marked as experimental.
Please, add `attrExperimental` to the `addChecker` call.